crates/countersyncd/benches/end_to_end.rs

```diff
@@ -0,0 +1,304 @@
+fn start_mock_collector() -> (String, oneshot::Sender<()>, thread::JoinHandle<()>, Arc<AtomicU64>) {
+    let (addr_tx, addr_rx) = std::sync::mpsc::channel();
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
+    let exports = Arc::new(AtomicU64::new(0));
+    let exports_clone = exports.clone();
+
+    let handle = thread::spawn(move || {
+        let rt = tokio::runtime::Runtime::new().expect("mock collector runtime");
+        rt.block_on(async move {
+            let listener = tokio::net::TcpListener::bind("127.0.0.1:0")
+                .await
+                .expect("bind mock collector");
+            let addr = listener.local_addr().expect("collector addr");
+            addr_tx.send(addr).expect("send collector addr");
+
+            let svc = MockMetricsService { exports: exports_clone };
+            let incoming = TcpListenerStream::new(listener);
+
+            Server::builder()
+                .add_service(
+                    opentelemetry_proto::tonic::collector::metrics::v1::metrics_service_server::MetricsServiceServer::new(
+                        svc,
+                    ),
+                )
+                .serve_with_incoming_shutdown(incoming, async {
+                    let _ = shutdown_rx.await;
+                })
+                .await
+                .ok();
+        });
+    });
+
+    let addr: SocketAddr = addr_rx.recv().expect("collector addr recv");
+    (format!("http://{}", addr), shutdown_tx, handle, exports)
+}
+
+fn flush_counters_db() {
+    let output = Command::new("redis-cli")
+        .args([
+            "-s",
+            SOCK_PATH,
+            "-n",
+            &COUNTERS_DB_ID.to_string(),
+            "FLUSHDB",
+        ])
+        .stdout(Stdio::null())
+        .output()
+        .expect("spawn redis-cli for flush");
+
+    if !output.status.success() {
+        panic!("redis-cli FLUSHDB failed with status {}", output.status);
+    }
+}
+
+fn seed_port_name_map(port_count: usize) {
+    let db = swss_common::DbConnector::new_unix(COUNTERS_DB_ID, SOCK_PATH, 0)
+        .expect("connect counter db for seed");
+
+    let table = "COUNTERS_PORT_NAME_MAP";
+    for idx in 0..port_count {
+        let name = format!("Ethernet{}", idx);
+        let value = swss_common::CxxString::from(format!("oid:0x100000000{:04x}", idx));
+        let _ = db.hset(table, name.as_str(), &*value);
+    }
+}
+
+async fn run_end_to_end(prepared: PreparedDataset, endpoint: String, exports_counter: Arc<AtomicU64>) -> (Duration, usize, u64) {
+    let (template_tx, template_rx) = mpsc::channel::<IPFixTemplatesMessage>(prepared.template_messages.len() + 4);
+    let (buffer_tx, buffer_rx) = mpsc::channel::<SocketBufferMessage>(1024);
+    let (counter_tx, counter_rx) = mpsc::channel::<SAIStatsMessage>(1024);
+    let (otel_tx, otel_rx) = mpsc::channel::<SAIStatsMessage>(1024);
+    let (stats_tx, stats_rx) = mpsc::channel::<SAIStatsMessage>(1024);
+    let (otel_done_tx, otel_done_rx) = oneshot::channel();
```

Comment:
do we have any log to monitor the actor queue length? this seems to be important from monitor point of view so we know how busy everything is easily.